draft-ietf-webpush-protocol.xml

```diff
@@ -565,12 +565,11 @@ TTL = 1*DIGIT
           TTL header field value.
         </t>
         <t>
-          A Push message with a zero TTL is immediately delivered if the user agent is
-          available to receive the message. After delivery, the push service is permitted
-          to immediately remove a push message with a zero TTL. This might occur before the
-          user agent acknowledges receipt of the message by performing a HTTP DELETE on the
-          push message resource. Consequently, an application server cannot rely on receiving
-          acknowledgement receipts for zero TTL push messages.
+          A push service MUST NOT generate a delivery receipt for a push message

```

Comment:
A push service may still deliver a receipt, even if they don't store the message - DELETE will use the message id ( resource URL ), which can encode enough info to generate the receipt. 
Or the push service may chose to hold enough info to generate the receipt.